src/Microsoft.Win32.SystemEvents/src/Microsoft/Win32/SessionEndingEventArgs.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.Win32
+{
+    /// <devdoc>
+    /// <para>Provides data for the <see cref='Microsoft.Win32.SystemEvents.SessionEnding'/> event.</para>
+    /// </devdoc>
+    public class SessionEndingEventArgs : EventArgs
+    {
+        private bool _cancel;
+        private readonly SessionEndReasons _reason;
+
+        /// <devdoc>
+        /// <para>Initializes a new instance of the <see cref='Microsoft.Win32.SessionEndingEventArgs'/> class.</para>
+        /// </devdoc>
+        public SessionEndingEventArgs(SessionEndReasons reason)
+        {
+            _reason = reason;
+        }
+
+        /// <devdoc>
+        ///    <para>Gets or sets a value indicating whether to cancel the user request to end the session.</para>
+        /// </devdoc>
+        public bool Cancel
```

Comment:
I guess it's a non goal to change this to {get; set; } and suchlike in this PR

Reply:
Yeah, I tried to keep the diff's over the ported code to a minimum.